src/clarity-angular/datagrid/datagrid-column-toggle.spec.ts

```diff
@@ -0,0 +1,306 @@
+/*
+ * Copyright (c) 2016-2017 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+import { Component, QueryList, TemplateRef, ViewChildren } from "@angular/core";
+import { TestContext } from "./helpers.spec";
+import { HideableColumnService } from "./providers/hideable-column.service";
+import { DatagridColumnToggle } from "./datagrid-column-toggle";
+import { DatagridHideableColumn } from "./datagrid-hideable-column";
+
+export default function (): void {
+    describe("Datagrid Column Toggle component", function () {
+        describe("Typescript API", function () {
```

Comment:
fyi - it is preserving the widths when I hide/show after dragging tho ... 
![out](https://cloud.githubusercontent.com/assets/433692/25584012/25a0c636-2e49-11e7-9287-8e74d5e48fcf.gif)